### PR TITLE
feature: Add /api/v1/users/_login endpoint

### LINKED ISSUE #51: Implement Login Endpoint
As a business user I want to persist login credentials  so that valid users can access their own account information and have access to functionalities on the website.

Tasks:
- [x] Update Users to hold Account info [Kevin]
- [x] Update Users controller for saving Account info [Alec]
- [x] Create controller for getting Login info [Kevin, Alec]
- [x] Create the route for POST /login [3dxp]

Acceptance Criteria:

- When a POST request is made against the /v1/users endpoint with a JSON request body with a first name, last name, username, and Account info. A response with the location header set with the relative URL of the created user with a success status code is returned to the client.
- When a POST request is made against the /login endpoint, the login credential is authenticated and the if valid will return a login "token".
- When an invalid Login credentials are given an error code 401 is returned.

### PR DESCRIPTION
**Pull request by 3dxp**
Create the endpoint for getting a JWT when providing
a username and password.

#51

### CODE REVIEW

**Comment by 3dxp** [comment]
This PR is dependent on work in #90, so please review that before this.

File: packages/hackmcx-api/src/utils/validation.js [hunk 1]
```diff
@@ -0,0 +1,5 @@
+const whitespaceRegex = /^\s*$/.compile();
```
Comment (acampb65) [inline packages/hackmcx-api/src/utils/validation.js]:
```suggestion
const whitespaceRegex = new RegExp("/^\s*$/");
```
Apparently `RegExp.compile()` is deprecated and actually didn't work properly on my machine. If I'm not mistaken using the constructor instead gives us the same functionality and did end up working on my end at least.

**Review by acampb65 — CHANGES_REQUESTED** [review]

File: packages/hackmcx-api/src/utils/validation.js (hunk 1 above)
Comment (acampb65) [reply packages/hackmcx-api/src/utils/validation.js]: `whitespaceRegex = /^\s*$/` works as well. 
`whitespaceRegex = new RegExp("/^\s*$/")` would be redundant, I think.

**Review by zxkevin — DISMISSED** [review]

**Review by acampb65 — DISMISSED** [review]

File: packages/hackmcx-api/src/utils/validation.js (hunk 1 above)
Comment (acampb65) [inline packages/hackmcx-api/src/utils/validation.js]:
```suggestion
const whitespaceRegex = new RegExp("/^\s*$/");
```

**Review by acampb65 — CHANGES_REQUESTED** [review]

**Review by acampb65 — APPROVED** [review]

**Review by zxkevin — APPROVED** [review]

**Pull request merged**